openshift/release/update-faas.sh

```diff
@@ -0,0 +1,87 @@
+#!/bin/bash
+
+
+readonly ROOT_DIR=$(dirname $0)/../..
+
+FAAS_VERSION=${1:-"main"}
+FAAS_REPO="github.com/boson-project/faas"
+
+UPDATED_FILES=$(cat <<EOT | tr '\n' ' '
+vendor
+pkg/kn/root/plugin_register.go
+go.mod
+go.sum
+EOT
+)
+
+
+generate_file() {
+  local faas_repo=$1
+  
+  cat <<EOF > "${ROOT_DIR}/pkg/kn/root/plugin_register.go"
+// Copyright © 2020 The Knative Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package root
+
+import (
+	_ "${faas_repo}/plugin"
+)
+
+// RegisterInlinePlugins is an empty function which however forces the
+// compiler to run all init() methods of the registered imports
+func RegisterInlinePlugins() {}
+EOF
+}
+
+mod_replace() {
+  local faas_version=$1
+  
+  cat <<EOF >> "${ROOT_DIR}/go.mod"
+replace (
+    github.com/boson-project/faas => github.com/boson-project/faas ${faas_version}
+
+    // Pin conflicting dependency versions
+    // Buildpacks required version
+    github.com/docker/docker => github.com/docker/docker v1.4.2-0.20200221181110-62bd5a33f707
+    // Darwin cross-build required version
+    golang.org/x/sys => golang.org/x/sys v0.0.0-20200302150141-5c8b2ff67527
+)
+EOF
+}
+
+mod_update() {
+  go mod tidy
+  go mod vendor
```

Comment:
IIRC upstream there is an additional step that 'cleans up' the vendor directory by removing LICENSE files and what not (or collecting license files). Maybe we should do this here, too ?

Reply:
I can add that for sure.

```bash
  # Cleanup
  find "./vendor" \( -name "OWNERS" -o -name "*_test.go" \) -print0 | xargs -0 rm -f
```

In our case it's only cleaning `OWNERS` file:
<details>
<summary>Cleanup preview</summary>

```
Changes not staged for commit:
  (use "git add/rm <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	deleted:    vendor/k8s.io/apimachinery/pkg/api/errors/OWNERS
	deleted:    vendor/k8s.io/apimachinery/pkg/api/meta/OWNERS
	deleted:    vendor/k8s.io/apimachinery/pkg/api/resource/OWNERS
	deleted:    vendor/k8s.io/apimachinery/pkg/apis/meta/v1/OWNERS
	deleted:    vendor/k8s.io/apimachinery/pkg/util/mergepatch/OWNERS
	deleted:    vendor/k8s.io/apimachinery/pkg/util/strategicpatch/OWNERS
	deleted:    vendor/k8s.io/apimachinery/third_party/forked/golang/json/OWNERS
	deleted:    vendor/k8s.io/client-go/pkg/apis/clientauthentication/OWNERS
	deleted:    vendor/k8s.io/client-go/plugin/pkg/client/auth/gcp/OWNERS
	deleted:    vendor/k8s.io/client-go/plugin/pkg/client/auth/oidc/OWNERS
	deleted:    vendor/k8s.io/client-go/rest/OWNERS
	deleted:    vendor/k8s.io/client-go/tools/auth/OWNERS
	deleted:    vendor/k8s.io/client-go/tools/cache/OWNERS
	deleted:    vendor/k8s.io/client-go/tools/metrics/OWNERS
	deleted:    vendor/k8s.io/client-go/transport/OWNERS
	deleted:    vendor/k8s.io/client-go/util/cert/OWNERS
	deleted:    vendor/k8s.io/client-go/util/keyutil/OWNERS
	deleted:    vendor/k8s.io/client-go/util/retry/OWNERS
	deleted:    vendor/k8s.io/code-generator/OWNERS
	deleted:    vendor/k8s.io/code-generator/cmd/client-gen/OWNERS
	deleted:    vendor/k8s.io/code-generator/cmd/go-to-protobuf/OWNERS
	deleted:    vendor/k8s.io/klog/OWNERS
	deleted:    vendor/k8s.io/klog/v2/OWNERS
	deleted:    vendor/k8s.io/kube-openapi/pkg/generators/rules/OWNERS
	deleted:    vendor/k8s.io/kube-openapi/pkg/util/proto/OWNERS
	deleted:    vendor/knative.dev/pkg/apis/OWNERS
	deleted:    vendor/knative.dev/pkg/apis/duck/OWNERS
	deleted:    vendor/knative.dev/pkg/configmap/OWNERS
	deleted:    vendor/knative.dev/pkg/kmeta/OWNERS
	deleted:    vendor/knative.dev/pkg/logging/OWNERS
	deleted:    vendor/knative.dev/pkg/test/OWNERS
	deleted:    vendor/knative.dev/pkg/tracing/config/OWNERS
	deleted:    vendor/knative.dev/serving/pkg/apis/autoscaling/OWNERS
	deleted:    vendor/sigs.k8s.io/yaml/OWNERS

```
</details>